packages/connect-core/src/entities/Organization.ts

```diff
@@ -81,102 +116,63 @@ export default class Organization {
   }
 
   ///////// APPS ///////////
-  async apps(): Promise<App[]> {
+
+  async app(filters?: AppFiltersParam): Promise<App> {
     this.checkConnected()
-    return this._connector.appsForOrg(this.address)
+    return this._connector.appForOrg(this.address, normalizeAppFilters(filters))
   }
 
-  onApps(callback: (apps: App[]) => void): { unsubscribe: Function } {
+  async apps(filters?: AppFiltersParam): Promise<App[]> {
     this.checkConnected()
-    return this._connector.onAppsForOrg(this.address, callback)
+    return this._connector.appsForOrg(
+      this.address,
+      normalizeAppFilters(filters)
+    )
   }
 
-  async app(appAddress: string): Promise<App> {
+  onApp(
+    filtersOrCallback: AppFiltersParam | OnAppCallback,
+    callback?: OnAppCallback
+  ): SubscriptionHandler {
     this.checkConnected()
-    return this._connector.appByAddress(appAddress)
-  }
-
-  // async addApp(
-  //   repoName: string,
-  //   {
-  //     initFuncName,
-  //     initFuncArgs,
-  //     openPermissions,
-  //   }: {
-  //     initFuncName: string
-  //     initFuncArgs: string[]
-  //     openPermissions: boolean
-  //   }
-  // ): Promise<TransactionIntent> {
-  //   return []
-  // }
-
-  // async removeApp(appAddress: string): Promise<TransactionIntent> {
-  //   return []
-  // }
+
+    const filters = (callback ? filtersOrCallback : null) as AppFiltersParam
+    const _callback = (callback || filtersOrCallback) as OnAppCallback
+
+    return this._connector.onAppForOrg(
+      this.address,
+      normalizeAppFilters(filters),
+      _callback
+    )
+  }
+
+  onApps(
+    filtersOrCallback: AppFiltersParam | OnAppsCallback,
+    callback?: OnAppsCallback
+  ): SubscriptionHandler {
+    this.checkConnected()
+
+    const filters = (callback ? filtersOrCallback : null) as AppFiltersParam
```

Comment:
Why the `filters` are only applied when a callback exists?

Reply:
This is to make the first parameter optional, so that we could do this:

```tsx
org.onApps('voting', apps => {
})
```

And this:

```tsx
org.onApps(apps => {
})
```

Reply:
Nice, now I understood the semantic of the syntax.